clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/KafkaConsumerServiceDelegator.java

```diff
@@ -124,45 +153,162 @@ public void startConsumptionIntoDataReceiver(
+    if (kafkaConsumerService != null) {
+      return kafkaConsumerService.getOffsetLagBasedOnMetrics(versionTopic, pubSubTopicPartition);
+    }
+    return -1;
   }
 
   @Override
   public long getLatestOffsetBasedOnMetrics(PubSubTopic versionTopic, PubSubTopicPartition pubSubTopicPartition) {
-    return getKafkaConsumerService(versionTopic, pubSubTopicPartition)
-        .getLatestOffsetBasedOnMetrics(versionTopic, pubSubTopicPartition);
+    KafkaConsumerService kafkaConsumerService = getKafkaConsumerService(versionTopic, pubSubTopicPartition);
+    if (kafkaConsumerService != null) {
+      return kafkaConsumerService.getLatestOffsetBasedOnMetrics(versionTopic, pubSubTopicPartition);
+    }
+    return -1;
   }
 
   @Override
   public Map<PubSubTopicPartition, TopicPartitionIngestionInfo> getIngestionInfoFromConsumer(
       PubSubTopic versionTopic,
       PubSubTopicPartition pubSubTopicPartition) {
-    return getKafkaConsumerService(versionTopic, pubSubTopicPartition)
-        .getIngestionInfoFromConsumer(versionTopic, pubSubTopicPartition);
+    KafkaConsumerService kafkaConsumerService = getKafkaConsumerService(versionTopic, pubSubTopicPartition);
+    if (kafkaConsumerService != null) {
+      return kafkaConsumerService.getIngestionInfoFromConsumer(versionTopic, pubSubTopicPartition);
+    }
+    return Collections.emptyMap();
   }
 
   @Override
   public boolean startInner() throws Exception {
-    defaultConsumerService.start();
-    if (consumerServiceForAAWCLeader != null) {
-      consumerServiceForAAWCLeader.start();
-    }
+    consumerServices.forEach(KafkaConsumerService::start);
     return true;
   }
 
   @Override
   public void stopInner() throws Exception {
-    defaultConsumerService.stop();
-    if (consumerServiceForAAWCLeader != null) {
-      consumerServiceForAAWCLeader.stop();
+    for (KafkaConsumerService kafkaConsumerService: consumerServices) {
+      kafkaConsumerService.stop();
+    }
+  }
+
+  interface ConsumerPoolStrategy {
+    KafkaConsumerService delegateKafkaConsumerServiceFor(
+        PubSubTopic versionTopic,
+        PartitionReplicaIngestionContext topicPartitionReplicaRole);
+
+    List<KafkaConsumerService> getConsumerServices();
+  }
+
+  public class DefaultConsumerPoolStrategy implements ConsumerPoolStrategy {
+    protected final KafkaConsumerService defaultConsumerService;
+
+    public DefaultConsumerPoolStrategy() {
+      defaultConsumerService = consumerServiceConstructor.apply(serverConfig.getConsumerPoolSizePerKafkaCluster(), "");
+    }
+
+    @Override
+    public KafkaConsumerService delegateKafkaConsumerServiceFor(
+        PubSubTopic versionTopic,
+        PartitionReplicaIngestionContext topicPartitionReplicaRole) {
+      return defaultConsumerService;
     }
+
+    @Override
+    public List<KafkaConsumerService> getConsumerServices() {
+      return Collections.singletonList(defaultConsumerService);
```

Comment:
Shall we maintain a list as class-level field, so that we don't need to construct a new list for every call?